homeassistant/components/backblaze/config_flow.py

```diff
@@ -0,0 +1,121 @@
+"""Config flow for the Backblaze integration."""
+
+from __future__ import annotations
+
+import logging
+from typing import Any, cast
+
+from b2sdk.v2 import B2Api, InMemoryAccountInfo, exception
+import voluptuous as vol
+
+from homeassistant.config_entries import ConfigFlow, ConfigFlowResult
+from homeassistant.helpers import config_validation as cv
+from homeassistant.helpers.selector import (
+    TextSelector,
+    TextSelectorConfig,
+    TextSelectorType,
+)
+
+from .const import CONF_APPLICATION_KEY, CONF_BUCKET, CONF_KEY_ID, CONF_PREFIX, DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+STEP_USER_DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_KEY_ID): cv.string,
+        vol.Required(CONF_APPLICATION_KEY): TextSelector(
+            config=TextSelectorConfig(type=TextSelectorType.PASSWORD)
+        ),
+        vol.Required(CONF_BUCKET): cv.string,
+        vol.Optional(CONF_PREFIX, default=""): cv.string,
+    }
+)
+
+
+class BackblazeConfigFlow(ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Backblaze."""
+
+    VERSION = 1
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Handle a flow initiated by the user."""
+        errors: dict[str, str] = {}
+        placeholders: dict[str, str] = {}
+
+        if user_input is not None:
+            self._async_abort_entries_match(
+                {
+                    CONF_KEY_ID: user_input[CONF_KEY_ID],
+                    CONF_APPLICATION_KEY: user_input[CONF_APPLICATION_KEY],
+                }
+            )
+
+            info = InMemoryAccountInfo()
+            b2_api = B2Api(info)
+
+            try:
+                _LOGGER.info(
+                    "Connecting to Backblaze with application key id %s",
+                    user_input[CONF_KEY_ID],
+                )
```

Comment:
Let's not log on info for this, ideally remove them or use `debug`

Reply:
Removed